test/solidity/blind-auction/blindAuction.js

```diff
@@ -0,0 +1,205 @@
+  const hundredHbars = 10000000000;
+  const twoHundredHbars = 20000000000;
+
+  before(async function () {
+    [beneficiary, wallet1] = await ethers.getSigners();
+  })
+
+  it('should confirm beneficiary is set correctly', async function () {
+    const contract = await deployBlindAuctionContract(3, 5, beneficiary.address);
+    const beneficiaryAddress = await contract.beneficiary();
+
+    expect(beneficiaryAddress).to.eq(beneficiary.address);
+  })
+
+  it('should confirm a user can bid', async function () {
+    const contract = await deployBlindAuctionContract(3, 5, beneficiary.address);
+
+    const bidData = ethers.utils.solidityKeccak256(["uint256", "bool", "uint256"] ,[hundred, false, 2]);
+
+    const result = await contract.connect(wallet1).bid(bidData, {value: hundred});
+    await result.wait();
+    const firstBidder = await contract.getBids(wallet1.address);
+
+    expect(firstBidder.length).to.eq(1);
+    expect(firstBidder[0].blindedBid).to.eq(bidData);
+  })
+
+  it('should confirm a user can reveal their bids', async function () {
+    const contract = await deployBlindAuctionContract(6, 5, beneficiary.address);
+
+    const firstBid = ethers.utils.solidityKeccak256(["uint256", "bool", "uint256"] ,[hundredHbars, false, ethers.utils.formatBytes32String('2')]);
+    const secondBid = ethers.utils.solidityKeccak256(["uint256", "bool", "uint256"] ,[hundredHbars, true, ethers.utils.formatBytes32String('23')]);
+
+    const bid = await contract.connect(wallet1).bid(firstBid, {value: hundred});
+    await bid.wait();
+
+    await sleep(2000);
+
+    const bid2 = await contract.connect(wallet1).bid(secondBid, {value: fiftyGwei});
+    await bid2.wait();
+
+    await sleep(3000);
+
+    const result = await contract.connect(wallet1).reveal([hundredHbars, hundredHbars], [false, true], [ethers.utils.formatBytes32String('2'), ethers.utils.formatBytes32String('23')], {gasLimit: 5000000});
+    await result.wait();
+    await sleep(3000);
+
+    const highestBidder =  await contract.highestBidder();
+    const highestBid = await contract.highestBid()
+
+    expect(highestBid).to.equal(BigInt(hundredHbars));
+    expect(highestBidder).to.equal(wallet1.address);
+  })
+
+  it('should confirm a user can withdraw', async function () {
+    const contract = await deployBlindAuctionContract(9, 5, beneficiary.address);
+
+    const firstBid = ethers.utils.solidityKeccak256(["uint256", "bool", "uint256"] ,[hundredHbars, false, ethers.utils.formatBytes32String('2')]);
+    const secondBid = ethers.utils.solidityKeccak256(["uint256", "bool", "uint256"] ,[twoHundredHbars, true, ethers.utils.formatBytes32String('23')]);
+    const thirdBid = ethers.utils.solidityKeccak256(["uint256", "bool", "uint256"] ,[twoHundredHbars, false, ethers.utils.formatBytes32String('5')]);
+
+    const bid = await contract.connect(wallet1).bid(firstBid, {value: hundred});
+    await bid.wait();
+
+    await sleep(2000);
```

Comment:
Why are we sleeping here and in all other places? Shouldn't wait be sufficient?

Reply:
Agreed.

Reply:
Let's not sleep as it's usually a sign we're missing something.
Waiting on the transaction hash should ensure the transaction has hit the mirror node and that the state changes it's caused are persisted.
If that's not the case let's investigate the issue

Reply:
@georgi-l95  @Nana-EC  @natanasow 
Okay so I removed the sleeps in this test except for the one here -> https://github.com/hashgraph/hedera-smart-contracts/blob/eaef6fe553e87f2b3c653b6a78a074094018d686/test/solidity/blind-auction/blindAuction.js#L116 
Which i reduced to 1 second - we need it because sometimes bidding takes less than the seconds I have set for biddingEnd so it will fail
The other sleep that seems necessary is the one after the withdraw https://github.com/hashgraph/hedera-smart-contracts/blob/eaef6fe553e87f2b3c653b6a78a074094018d686/test/solidity/blind-auction/blindAuction.js#L131
Behaviour is inconsitent if i remove it sometimes the first expect fails apparently because the balance has not been updated. When I add sleep of 1 seconds it always passes

Reply:
I still don't like the sleep. Every transaction submitted to the relay can be awaited and it ensures the transaction went E2E so there should be no need for sleeping.
Are we not awaiting on the right execution

Reply:
I think i've seen the same issue with the balances somewhere else, could this be a more complex bug with the relay itself ?

Reply:
Can we open a ticket to circle back and remove these sleeps

Reply:
@Nana-EC I removed the ones that are not unecessary, the ones I have left out are needed
It is part of the business logic of the contract. Is there another way to wait for a few block to pass?